src/components/sub-menu/SubMenu.vue

```diff
@@ -1,48 +1,46 @@
 <template>
   <v-menu
     :close-on-content-click="false"
-    :offset-x="isOffsetX"
-    :offset-y="isOffsetY"
     :open-on-hover="isOpenOnHover"
     :transition="transition"
-    :value="openMenu"
+    :model-value="openMenu"
```

Comment:
So this works and it's the update event that doesn't work?

Reply:
Yes it appears that the update event was mainly causing the issues. Ketaki's solution above seems to be the fix.